### DISCUSSION TITLE
Multiple ways to install / update nuclei

### DISCUSSION THREAD

**Discussion by ehsandeep** [projectdiscovery/nuclei#3679] [Show and tell]
## Using [PDTM](https://github.com/projectdiscovery/pdtm) (preferred way)
```bash
pdtm -i nuclei -u nuclei  # to install / update
```

## Using Nuclei (self update)

```bash
nuclei -update # to update (not available in kali os)
```

## Using Go (install)

```bash
go install -v github.com/projectdiscovery/nuclei/v2/cmd/nuclei@latest # to install / update
```

## Using Go (build)

```bash
git clone https://github.com/projectdiscovery/nuclei.git; \
cd nuclei/v2; \
make build; \
./nuclei -version;
```

## Using Brew
```bash
brew install nuclei # to install
brew upgrade nuclei # to update
```

## Using Docker
```bash
docker pull projectdiscovery/nuclei:latest # to install / update
```

**Comment by ghost** [comment]
am trying to update to the latest version of nuclei but am getting this error message every now and then 
[INF] Current nuclei version: v2.9.4 (outdated)
 i have downloaded the latest nuclei release and dont know how to go about it..any help here.

**Reply by MetzinAround** [reply]
what method from above are you attempting to use?

**Reply by ghost** [reply]
i used the go install,then it said that i need go1.12 or such,,installed it and then still the error

**Reply by ghost** [reply]
![WhatsApp Image 2023-06-14 at 18 42 58](https://github.com/projectdiscovery/nuclei/assets/112280231/d262b4ec-caae-4671-8daf-f19bcdaeca7d)

**Reply by Mzack9999** [reply]
I suspect you are running the wrong binary. Please ensure that `$home/go/bin` is within the `$PATH` environment variable

**Reply by MetzinAround** [reply]
https://www.youtube.com/watch?v=wtHcXA2l9c4 has step by steps on installing using PDTM

**Reply by MetzinAround** [reply]
would `nuclei -update` work?

**Comment by ViCrack** [comment]
Using [Scoop](https://github.com/ScoopInstaller/Scoop)

`scoop install https://raw.githubusercontent.com/vicrack/scoop-bucket/master/bucket/nuclei.json`

`scoop update nuclei`

**Reply by sxzza** [reply]
[filexss.txt](https://github.com/user-attachments/files/17368953/filexss.txt)

**Comment by Rajiv-Stha** [comment]
Is the Homebrew version kept up to date with the latest releases here on GitHub?

**Reply by GeorginaReeder** [reply]
I believe so @Rajiv-Stha ! Homebrew lists v3.4.3 as the latest Nuclei version, which is correct: https://formulae.brew.sh/formula/

**Reply by Rajiv-Stha** [reply]
Thanks a lot!
really appreciate the quick response

**Comment by sSs-Lilith-sSs** [comment]
sudo cp ~/go/bin/nuclei /usr/local/bin/
